solidity/contracts/system/KeepFactorySelection.sol

```diff
@@ -101,78 +122,137 @@ library KeepFactorySelection {
         uint256 _groupSize,
         uint256 _honestThreshold
     ) public {
-        if (address(_self.keepStakeFactory) != address(0)) {
-            _self.keepStakeFactory.setMinimumBondableValue(
+        require(
+            address(_self.keepStakeVendor) != address(0),
+            "KEEP backed vendor not set"
+        );
+
+        IBondedECDSAKeepFactory keepStakeFactory = getKeepStakedFactory(_self);
+
+        if (address(keepStakeFactory) != address(0)) {
+            keepStakeFactory.setMinimumBondableValue(
                 _minimumBondableValue,
                 _groupSize,
                 _honestThreshold
             );
         }
-        if (address(_self.ethStakeFactory) != address(0)) {
-            _self.ethStakeFactory.setMinimumBondableValue(
-                _minimumBondableValue,
-                _groupSize,
-                _honestThreshold
+
+        if (address(_self.fullyBackedVendor) != address(0)) {
+            IBondedECDSAKeepFactory fullyBackedFactory = getFullyBackedFactory(
+                _self
             );
+
+            if (address(fullyBackedFactory) != address(0)) {
+                fullyBackedFactory.setMinimumBondableValue(
+                    _minimumBondableValue,
+                    _groupSize,
+                    _honestThreshold
+                );
+            }
         }
     }
 
-    /// @notice Refreshes the keep factory choice. If either ETH-stake factory
+    /// @notice Refreshes the keep factory choice. If either ETH-bond-only vendor
     /// or selection strategy is not set, KEEP-stake factory is selected.
     /// Otherwise, calls selection strategy providing addresses of both
     /// factories to make a choice. Additionally, passes the selection seed
     /// evaluated from the current request counter value.
+    /// Unless lock is set, it calls vendors to obtain the latests factories
+    /// versions.
     function refreshFactory(Storage storage _self) internal {
+        IBondedECDSAKeepFactory keepStakeFactory = getKeepStakedFactory(_self);
+
         if (
-            address(_self.ethStakeFactory) == address(0) ||
+            address(_self.fullyBackedVendor) == address(0) ||
             address(_self.factorySelector) == address(0)
         ) {
             // KEEP-stake factory is guaranteed to be there. If the selection
             // can not be performed, this is the default choice.
-            _self.selectedFactory = _self.keepStakeFactory;
+            _self.selectedFactory = keepStakeFactory;
```

Comment:
Can we make a small optimization and compare the value before it's set?

Reply:
I'm not sure what you mean. Compare to what?

Reply:
Compare to what's already in `_self.selectedFactory`. If the value stored in `_self.selectedFactory` is the same as the one returned from `getKeepStakedFactory(_self)`, we don't need to write to `_self.selectedFactory` saving gas on SSTORE operation.

Reply:
b9b3b57612e0774872a036c26c1005f2802c1c2d